### PR TITLE
Xdg thumbnailer

### PR DESCRIPTION
**Pull request by azubieta**
Initial version of the Xdg thumbnailer

It calls generate thumbnails every time that an appimage is regisred in the system.

### CODE REVIEW

**Comment by probonopd** [comment]
Thanks @azubieta. Strangely, we got 

```
curl: (7) Failed to connect to transfer.sh port 443: Connection timed out
```

So I retriggered the build in order to be able to try it out.

File: shared.c [hunk 1]
```diff
@@ -274,6 +278,69 @@ static gchar *get_file_extension(const gchar *filename)
...
+        char buf[bytes_at_a_time];
+        if (sqfs_read_range(fs, &inode, (sqfs_off_t) bytes_already_read, &bytes_at_a_time, buf))
+            fprintf(stderr, "sqfs_read_range error\n");
+        fwrite(buf, 1, bytes_at_a_time, f);
+        bytes_already_read = bytes_already_read + bytes_at_a_time;
+    }
+    fclose(f);
+}
+
+gchar * gess_install_destination(const gchar *desktop_icon_value_original, const char *path, const char *md5)
```
Comment (probonopd) [inline shared.c]: Should it mean "guess"?

Comment (azubieta) [reply shared.c]: yes, sorry for the typos I should get a spell checker in my IDE

Comment (azubieta) [reply shared.c]: I'm sending a new pr right away

Comment (probonopd) [reply shared.c]: np, thanks

**Comment by azubieta** [comment]
By the way despite of how perfectly we implement the XDG Thumbnails spec some file managers just ignore them, by example dolphin and konkeror.
https://wiki.archlinux.org/index.php/File_manager_functionality#Thumbnail_previews

**Comment by probonopd** [comment]
@TheAssassin do you have an idea why https://travis-ci.org/AppImage/AppImageKit/jobs/313581647#L8403 ff does not output the transfer.sh URLs anymore?

**Comment by TheAssassin** [comment]
@probonopd it says `Could not save metadata`, which is an indicator that they ran out of space. I'll retrigger the build, and see whether the problem has been fixed.

**Comment by probonopd** [comment]
Indeed, the links are there now. Thanks @TheAssassin

**Comment by probonopd** [comment]
This is what I am getting:

```
me@host ~ $ Downloads/appimaged-x86_64.AppImage 
_________________________

-> REGISTER /home/me/Downloads/appimaged-x86_64.AppImage
_________________________
AppImage type: 2
Opening /home/me/Downloads/appimaged-x86_64.AppImage as Type 2 AppImage
WARNING: Unable to extract non regular file ".DirIcon", Type 3
Closing /home/me/Downloads/appimaged-x86_64.AppImage
squashfs based type 2 AppImage
Got root desktop: appimaged.desktop
.upd_info offset: 95703
.upd_info length: 1024
*** Error in `appimaged': munmap_chunk(): invalid pointer: 0x000000000042770b ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f9aeff577e5]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x1a8)[0x7f9aeff63ae8]
appimaged[0x405bfc]
appimaged[0x406382]
appimaged[0x40707f]
appimaged[0x40709f]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba)[0x7f9af04ca6ba]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f9aeffe682d]
======= Memory map: ========
00400000-0043f000 r-xp 00000000 00:2f 133                                /tmp/.mount_appimadat3GX/usr/bin/appimaged
0063f000-00640000 rw-p 0003f000 00:2f 133                                /tmp/.mount_appimadat3GX/usr/bin/appimaged
00640000-00654000 rw-p 00000000 00:00 0 
01bd8000-01bf9000 rw-p 00000000 00:00 0                                  [heap]
7f9ae8000000-7f9ae8021000 rw-p 00000000 00:00 0 
7f9ae8021000-7f9aec000000 ---p 00000000 00:00 0 
7f9aec78c000-7f9aec7a2000 r-xp 00000000 07:01 12046                      /lib/x86_64-linux-gnu/libgcc_s.so.1
7f9aec7a2000-7f9aec9a1000 ---p 00016000 07:01 12046                      /lib/x86_64-linux-gnu/libgcc_s.so.1
7f9aec9a1000-7f9aec9a2000 rw-p 00015000 07:01 12046                      /lib/x86_64-linux-gnu/libgcc_s.so.1
7f9aec9a2000-7f9aec9a3000 ---p 00000000 00:00 0 
7f9aec9a3000-7f9aed1a3000 rw-p 00000000 00:00 0 
7f9aed1a3000-7f9aed1a8000 r-xp 00000000 07:01 39023                      /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f9aed1a8000-7f9aed3a7000 ---p 00005000 07:01 39023                      /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f9aed3a7000-7f9aed3a8000 r--p 00004000 07:01 39023                      /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f9aed3a8000-7f9aed3a9000 rw-p 00005000 07:01 39023                      /usr/lib/x86_64-linux-gnu/libXdmcp.so.6.0.0
7f9aed3a9000-7f9aed3ab000 r-xp 00000000 07:01 39012                      /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f9aed3ab000-7f9aed5ab000 ---p 00002000 07:01 39012                      /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f9aed5ab000-7f9aed5ac000 r--p 00002000 07:01 39012                      /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f9aed5ac000-7f9aed5ad000 rw-p 00003000 07:01 39012                      /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0
7f9aed5ad000-7f9aed5d3000 r-xp 00000000 07:01 12039                      /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f9aed5d3000-7f9aed7d3000 ---p 00026000 07:01 12039                      /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f9aed7d3000-7f9aed7d5000 r--p 00026000 07:01 12039                      /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f9aed7d5000-7f9aed7d6000 rw-p 00028000 07:01 12039                      /lib/x86_64-linux-gnu/libexpat.so.1.6.0
7f9aed7d6000-7f9aed8de000 r-xp 00000000 07:01 12078                      /lib/x86_64-linux-gnu/libm-2.23.so
7f9aed8de000-7f9aedadd000 ---p 00108000 07:01 12078                      /lib/x86_64-linux-gnu/libm-2.23.so
7f9aedadd000-7f9aedade000 r--p 00107000 07:01 12078                      /lib/x86_64-linux-gnu/libm-2.23.so
7f9aedade000-7f9aedadf000 rw-p 00108000 07:01 12078                      /lib/x86_64-linux-gnu/libm-2.23.so
7f9aedadf000-7f9aedae6000 r-xp 00000000 07:01 12162                      /lib/x86_64-linux-gnu/librt-2.23.so
7f9aedae6000-7f9aedce5000 ---p 00007000 07:01 12162                      /lib/x86_64-linux-gnu/librt-2.23.so
7f9aedce5000-7f9aedce6000 r--p 00006000 07:01 12162                      /lib/x86_64-linux-gnu/librt-2.23.so
7f9aedce6000-7f9aedce7000 rw-p 00007000 07:01 12162                      /lib/x86_64-linux-gnu/librt-2.23.so
7f9aedce7000-7f9aedcf8000 r-xp 00000000 07:01 39025                      /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f9aedcf8000-7f9aedef7000 ---p 00011000 07:01 39025                      /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f9aedef7000-7f9aedef8000 r--p 00010000 07:01 39025                      /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f9aedef8000-7f9aedef9000 rw-p 00011000 07:01 39025                      /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0
7f9aedef9000-7f9aee02e000 r-xp 00000000 07:01 39008                      /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f9aee02e000-7f9aee22e000 ---p 00135000 07:01 39008                      /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f9aee22e000-7f9aee22f000 r--p 00135000 07:01 39008                      /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f9aee22f000-7f9aee233000 rw-p 00136000 07:01 39008                      /usr/lib/x86_64-linux-gnu/libX11.so.6.3.0
7f9aee233000-7f9aee23c000 r-xp 00000000 07:01 39045                      /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f9aee23c000-7f9aee43b000 ---p 00009000 07:01 39045                      /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f9aee43b000-7f9aee43c000 r--p 00008000 07:01 39045                      /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f9aee43c000-7f9aee43d000 rw-p 00009000 07:01 39045                      /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0
7f9aee43d000-7f9aee45e000 r-xp 00000000 07:01 40591                      /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f9aee45e000-7f9aee65d000 ---p 00021000 07:01 40591                      /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f9aee65d000-7f9aee65e000 r--p 00020000 07:01 40591                      /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f9aee65e000-7f9aee65f000 rw-p 00021000 07:01 40591                      /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0
7f9aee65f000-7f9aee667000 r-xp 00000000 07:01 40577                      /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f9aee667000-7f9aee867000 ---p 00008000 07:01 40577                      /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f9aee867000-7f9aee868000 r--p 00008000 07:01 40577                      /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f9aee868000-7f9aee869000 rw-p 00009000 07:01 40577                      /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0
7f9aee869000-7f9aee86b000 r-xp 00000000 07:01 40581                      /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f9aee86b000-7f9aeea6b000 ---p 00002000 07:01 40581                      /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f9aeea6b000-7f9aeea6c000 r--p 00002000 07:01 40581                      /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f9aeea6c000-7f9aeea6d000 rw-p 00003000 07:01 40581                      /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
7f9aeea6d000-7f9aeea91000 r-xp 00000000 07:01 12149                      /lib/x86_64-linux-gnu/libpng12.so.0.54.0
7f9aeea91000-7f9aeec90000 ---p 00024000 07:01 12149                      /lib/x86_64-linux-gnu/libpng12.so.0.54.0
7f9aeec90000-7f9aeec91000 r--p 00023000 07:01 12149                      /lib/x86_64-linux-gnu/libpng12.so.0.54.0
7f9aeec91000-7f9aeec92000 rw-p 00024000 07:01 12149                      /lib/x86_64-linux-gnu/libpng12.so.0.54.0
7f9aeec92000-7f9aeed36000 r-xp 00000000 07:01 39464                      /usr/lib/x86_64-linux-gnu/libfreetype.so.6.12.1
7f9aeed36000-7f9aeef35000 ---p 000a4000 07:01 39464                      /usr/lib/x86_64-linux-gnu/libfreetype.so.6.12.1
7f9aeef35000-7f9aeef3b000 r--p 000a3000 07:01 39464                      /usr/lib/x86_64-linux-gnu/libfreetype.so.6.12.1
7f9aeef3b000-7f9aeef3c000 rw-p 000a9000 07:01 39464                      /usr/lib/x86_64-linux-gnu/libfreetype.so.6.12.1
7f9aeef3c000-7f9aeef79000 r-xp 00000000 07:01 39430                      /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.0
7f9aeef79000-7f9aef178000 ---p 0003d000 07:01 39430                      /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.0
7f9aef178000-7f9aef17a000 r--p 0003c000 07:01 39430                      /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.0
7f9aef17a000-7f9aef17f000 rw-p 0003e000 07:01 39430                      /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.9.0
7f9aef17f000-7f9aef21e000 r-xp 00000000 07:01 40151                      /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.33.6
7f9aef21e000-7f9aef41e000 ---p 0009f000 07:01 40151                      /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.33.6
7f9aef41e000-7f9aef426000 r--p 0009f000 07:01 40151                      /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.33.6
7f9aef426000-7f9aef427000 rw-p 000a7000 07:01 40151                      /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.33.6
7f9aef427000-7f9aef42e000 r-xp 00000000 07:01 39408                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f9aef42e000-7f9aef62d000 ---p 00007000 07:01 39408                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f9aef62d000-7f9aef62e000 r--p 00006000 07:01 39408                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f9aef62e000-7f9aef62f000 rw-p 00007000 07:01 39408                      /usr/lib/x86_64-linux-gnu/libffi.so.6.0.4
7f9aef62f000-7f9aef646000 r-xp 00000000 07:01 12160                      /lib/x86_64-linux-gnu/libresolv-2.23.so
7f9aef646000-7f9aef846000 ---p 00017000 07:01 12160                      /lib/x86_64-linux-gnu/libresolv-2.23.so
7f9aef846000-7f9aef847000 r--p 00017000 07:01 12160                      /lib/x86_64-linux-gnu/libresolv-2.23.so
7f9aef847000-7f9aef848000 rw-p 00018000 07:01 12160                      /lib/x86_64-linux-gnu/libresolv-2.23.so
7f9aef848000-7f9aef84a000 rw-p 00000000 00:00 0 
7f9aef84a000-7f9aef869000 r-xp 00000000 07:01 12166                      /lib/x86_64-linux-gnu/libselinux.so.1
7f9aef869000-7f9aefa68000 ---p 0001f000 07:01 12166                      /lib/x86_64-linux-gnu/libselinux.so.1
7f9aefa68000-7f9aefa69000 r--p 0001e000 07:01 12166                      /lib/x86_64-linux-gnu/libselinux.so.1
7f9aefa69000-7f9aefa6a000 rw-p 0001f000 07:01 12166                      /lib/x86_64-linux-gnu/libselinux.so.1
7f9aefa6a000-7f9aefa6c000 rw-p 00000000 00:00 0 
7f9aefa6c000-7f9aefa6f000 r-xp 00000000 07:01 39555                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.4800.1
7f9aefa6f000-7f9aefc6e000 ---p 00003000 07:01 39555                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.4800.1
7f9aefc6e000-7f9aefc6f000 r--p 00002000 07:01 39555                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.4800.1
7f9aefc6f000-7f9aefc70000 rw-p 00003000 07:01 39555                      /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.4800.1
7f9aefc70000-7f9aefcde000 r-xp 00000000 07:01 12137                      /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f9aefcde000-7f9aefede000 ---p 0006e000 07:01 12137                      /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f9aefede000-7f9aefedf000 r--p 0006e000 07:01 12137                      /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f9aefedf000-7f9aefee0000 rw-p 0006f000 07:01 12137                      /lib/x86_64-linux-gnu/libpcre.so.3.13.2
7f9aefee0000-7f9af009f000 r-xp 00000000 07:01 12008                      /lib/x86_64-linux-gnu/libc-2.23.so
7f9af009f000-7f9af029f000 ---p 001bf000 07:01 12008                      /lib/x86_64-linux-gnu/libc-2.23.so
7f9af029f000-7f9af02a3000 r--p 001bf000 07:01 12008                      /lib/x86_64-linux-gnu/libc-2.23.so
7f9af02a3000-7f9af02a5000 rw-p 001c3000 07:01 12008                      /lib/x86_64-linux-gnu/libc-2.23.so
7f9af02a5000-7f9af02a9000 rw-p 00000000 00:00 0 
7f9af02a9000-7f9af02c2000 r-xp 00000000 07:01 12199                      /lib/x86_64-linux-gnu/libz.so.1.2.8
7f9af02c2000-7f9af04c1000 ---p 00019000 07:01 12199                      /lib/x86_64-linux-gnu/libz.so.1.2.8
7f9af04c1000-7f9af04c2000 r--p 00018000 07:01 12199                      /lib/x86_64-linux-gnu/libz.so.1.2.8
7f9af04c2000-7f9af04c3000 rw-p 00019000 07:01 12199                      /lib/x86_64-linux-gnu/libz.so.1.2.8
7f9af04c3000-7f9af04db000 r-xp 00000000 07:01 12154                      /lib/x86_64-linux-gnu/libpthread-2.23.so
7f9af04db000-7f9af06da000 ---p 00018000 07:01 12154                      /lib/x86_64-linux-gnu/libpthread-2.23.so
7f9af06da000-7f9af06db000 r--p 00017000 07:01 12154                      /lib/x86_64-linux-gnu/libpthread-2.23.so
7f9af06db000-7f9af06dc000 rw-p 00018000 07:01 12154                      /lib/x86_64-linux-gnu/libpthread-2.23.so
7f9af06dc000-7f9af06e0000 rw-p 00000000 00:00 0 
7f9af06e0000-7f9af06e3000 r-xp 00000000 07:01 12032                      /lib/x86_64-linux-gnu/libdl-2.23.so
7f9af06e3000-7f9af08e2000 ---p 00003000 07:01 12032                      /lib/x86_64-linux-gnu/libdl-2.23.so
7f9af08e2000-7f9af08e3000 r--p 00002000 07:01 12032                      /lib/x86_64-linux-gnu/libdl-2.23.so
7f9af08e3000-7f9af08e4000 rw-p 00003000 07:01 12032                      /lib/x86_64-linux-gnu/libdl-2.23.so
7f9af08e4000-7f9af09f2000 r-xp 00000000 07:01 39175                      /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.6
7f9af09f2000-7f9af0bf2000 ---p 0010e000 07:01 39175                      /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.6
7f9af0bf2000-7f9af0bf5000 r--p 0010e000 07:01 39175                      /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.6
7f9af0bf5000-7f9af0bf6000 rw-p 00111000 07:01 39175                      /usr/lib/x86_64-linux-gnu/libcairo.so.2.11400.6
7f9af0bf6000-7f9af0bf8000 rw-p 00000000 00:00 0 
7f9af0bf8000-7f9af0c4a000 r-xp 00000000 07:01 39583                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4800.1
7f9af0c4a000-7f9af0e49000 ---p 00052000 07:01 39583                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4800.1
7f9af0e49000-7f9af0e4a000 r--p 00051000 07:01 39583                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4800.1
7f9af0e4a000-7f9af0e4b000 rw-p 00052000 07:01 39583                      /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4800.1
7f9af0e4b000-7f9af0fcb000 r-xp 00000000 07:01 39536                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.4800.1
7f9af0fcb000-7f9af11cb000 ---p 00180000 07:01 39536                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.4800.1
7f9af11cb000-7f9af11cf000 r--p 00180000 07:01 39536                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.4800.1
7f9af11cf000-7f9af11d1000 rw-p 00184000 07:01 39536                      /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.4800.1
7f9af11d1000-7f9af11d3000 rw-p 00000000 00:00 0 
7f9af11d3000-7f9af12e2000 r-xp 00000000 07:01 12050                      /lib/x86_64-linux-gnu/libglib-2.0.so.0.4800.1
7f9af12e2000-7f9af14e1000 ---p 0010f000 07:01 12050                      /lib/x86_64-linux-gnu/libglib-2.0.so.0.4800.1
7f9af14e1000-7f9af14e2000 r--p 0010e000 07:01 12050                      /lib/x86_64-linux-gnu/libglib-2.0.so.0.4800.1
7f9af14e2000-7f9af14e3000 rw-p 0010f000 07:01 12050                      /lib/x86_64-linux-gnu/libglib-2.0.so.0.4800.1
7f9af14e3000-7f9af14e4000 rw-p 00000000 00:00 0 
7f9af14e4000-7f9af150a000 r-xp 00000000 07:01 11980                      /lib/x86_64-linux-gnu/ld-2.23.so
7f9af167b000-7f9af16b9000 r--s 00000000 00:14 44167                      /home/me/Downloads/appimaged-x86_64.AppImage
7f9af16b9000-7f9af16e8000 rw-p 00000000 00:00 0 
7f9af1706000-7f9af1709000 rw-p 00000000 00:00 0 
7f9af1709000-7f9af170a000 r--p 00025000 07:01 11980                      /lib/x86_64-linux-gnu/ld-2.23.so
7f9af170a000-7f9af170b000 rw-p 00026000 07:01 11980                      /lib/x86_64-linux-gnu/ld-2.23.so
7f9af170b000-7f9af170c000 rw-p 00000000 00:00 0 
7ffcbf6ac000-7ffcbf6ce000 rw-p 00000000 00:00 0                          [stack]
7ffcbf6db000-7ffcbf6dd000 r--p 00000000 00:00 0                          [vvar]
7ffcbf6dd000-7ffcbf6df000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aborted
```

Seems like symlinks are not handled properly. In many AppImages, `.DirIcon` is a symlink to the real icon file. `appimaged` would need to extract the target of the symlink in this case @azubieta.

**Comment by azubieta** [comment]
@probonopd any did you have this done somewhere else ?

**Comment by probonopd** [comment]
Here:

https://github.com/AppImage/AppImageKit/blob/bc39d5a8aa572fce44db4ce874a7a87473978892/runtime.c#L389-L400

**Comment by TheAssassin** [comment]
We should try to merge #568, which, among other fixes, also resolves this issue.

**Comment by probonopd** [comment]
I hope that at least for some time we'll try to keep `build.sh` working?

**Comment by TheAssassin** [comment]
@probonopd did you even have a look at the code? `build.sh` has been updated to build with CMake. I can't tell whether it works with incremental builds at the moment however.

File: shared.c [hunk 2]
```diff
@@ -878,6 +897,8 @@ bool appimage_type2_register_in_system(char *path, gboolean verbose)
     return TRUE;
 }
 
+void create_thumbnail(const gchar * appimage_file_path, gboolean verbose);
```
Comment (TheAssassin) [inline shared.c]: That seems like a good idea, we could even narrow it further to `const char const* appimage_file_path`.

File: shared.c [hunk 3]
```diff
@@ -975,3 +989,356 @@ int appimage_unregister_in_system(char *path, gboolean verbose)
     g_free(md5);
     return 0;
 }
+
+
+int match_regexp(const gchar *text, const gchar *pattern)
```
Comment (TheAssassin) [inline shared.c]: That seems like a good idea, we could even narrow it further to `const char const* text, const gchar const* pattern`.

File: shared.c [hunk 4]
```diff
@@ -975,3 +989,356 @@ int appimage_unregister_in_system(char *path, gboolean verbose)
...
+    regmatch_t match[2];
+    regcomp(&regex, pattern, REG_ICASE | REG_EXTENDED);
+    result = regexec(&regex, text, 2, match, 0);
+    regfree(&regex);
+
+    return result;
+}
+
+
+/* AppImage generic handler calback to be used in algorithms */
```
Comment (TheAssassin) [inline shared.c]: Typo: it's `callback`.

File: shared.c [hunk 5]
```diff
@@ -975,3 +989,356 @@ int appimage_unregister_in_system(char *path, gboolean verbose)
...
+            sqfs_destroy(fs);
+        } else {
+            handler->is_open = true;
+            handler->cache = fs;
+        }
+    }
+}
+
+void appimage_type2_close(appimage_handler * handler) {
+    if ( is_handler_valid(handler) && handler->is_open ) {
```
Comment (TheAssassin) [inline shared.c]: Code style: `if (is_...` or `if(is...)`. I've spotted this in many of the new functions.

Comment (azubieta) [reply shared.c]: What do you exactly propose here ?

Comment (TheAssassin) [reply shared.c]: The space inside the bracket isn't really used in the rest of the file, as far as I can see. But I don't really mind the current state either. Personally, I prefer `if (...)` over `if(...)`, but the latter is the pattern used in the majority of the code files.

**Comment by azubieta** [comment]
closed in favor of #594

**Pull request closed without merging**